[FLINK-23513] Remove legacy descriptors

Airblader opened the pull request:
## What is the purpose of the change

This removes a lot of the legacy descriptors used for `#connect`, as well as for connectors/formats along with their legacy implementations.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

twalthr reviewed (commented):
Thanks for removing all this legacy. I added some feedback. Maybe we could add a couple of commits to remove even more descriptors which are not needed anymore:

- FunctionService
- FunctionDescriptor
- HierarchyDescriptor

Also we should deprecate Schema, Rowtime and the related sub graph. Ideally, the `o.a.f.table.descriptors` module should only contain deprecated classes in the end.

Airblader commented:
Thanks for the review! I've addressed all feedback as fixup commits placed after the corresponding commit, and added a few new commits removing even more and deprecating everything in the package that couldn't quite be removed yet.

twalthr reviewed (approved):
Great to drop even more :) LGTM
